src/Route.php

```diff
@@ -170,10 +174,30 @@ public function defaults(array $defaults): self
 
     /**
      * @param callable|MiddlewareInterface $middleware
-     * @return MiddlewareInterface
+     * @return MiddlewareInterface|string|array
      */
-    private function prepareMiddleware($middleware): MiddlewareInterface
+    private function prepareMiddleware($middleware)
     {
+        if (
+            is_string($middleware) && is_subclass_of($middleware, MiddlewareInterface::class)
+        ) {
+            if ($this->container === null) {
+                throw new InvalidArgumentException('Route container must not be null for lazy loaded middleware.');
+            }
+            return $middleware;
+        }
+
+        if (
```

Comment:
Can it be replaced with `is_callable` :question:

Reply:
No